src/applications/bmqstoragetool/m_bmqstoragetool_journalfileprocessor.t.cpp

```diff
@@ -2026,6 +2058,79 @@ static void test23_searchJournalOpRecordsBySeqNumber()
     BMQTST_ASSERT_EQ(resultStream.str(), expectedStream.str());
 }
 
+static void test24_summaryWithQueueDetailsTest()
+// ------------------------------------------------------------------------
+// OUTPUT SUMMARY TEST
+//
+// Concerns:
+//   Search messages in journal file and output summary.
+//
+// Testing:
+//   JournalFileProcessor::process()
+// ------------------------------------------------------------------------
+{
+    bmqtst::TestHelper::printTestName(
+        "OUTPUT SUMMARY WITH QUEUE DETAILS TEST");
+
+    // Simulate journal file
+    const size_t                 k_NUM_RECORDS = 15;
+    JournalFile::RecordsListType records(bmqtst::TestHelperUtil::allocator());
+    JournalFile                  journalFile(k_NUM_RECORDS,
+                            bmqtst::TestHelperUtil::allocator());
+    JournalFile::GuidVectorType  partiallyConfirmedGUIDS(
+        bmqtst::TestHelperUtil::allocator());
+    journalFile.addJournalRecordsWithPartiallyConfirmedMessages(
+        &records,
+        &partiallyConfirmedGUIDS);
+
+    // Configure parameters to output summary
+    Parameters params = createTestParameters();
+
+    params.d_summary            = true;
+    params.d_minRecordsPerQueue = 0;
+
+    params.d_processRecordTypes.d_message = true;
+
+    // Prepare file manager
+    bslma::ManagedPtr<FileManager> fileManager(
+        new (*bmqtst::TestHelperUtil::allocator())
+            FileManagerMock(journalFile),
+        bmqtst::TestHelperUtil::allocator());
+
+    // Run search
+    bmqu::MemOutStream resultStream(bmqtst::TestHelperUtil::allocator());
+    bslma::ManagedPtr<CommandProcessor> searchProcessor =
+        CommandProcessorFactory::createCommandProcessor(
+            &params,
+            fileManager,
+            resultStream,
+            bmqtst::TestHelperUtil::allocator());
+    searchProcessor->process();
+
+    // Prepare expected output
+    bmqu::MemOutStream expectedStream(bmqtst::TestHelperUtil::allocator());
+    expectedStream << "5 message(s) found.\n";
+    bsl::vector<const char*> fields(bmqtst::TestHelperUtil::allocator());
```

Comment:
Not needed to fix now, but I opened an issue for this thing
https://github.com/bloomberg/blazingmq/issues/564